chemprop/uncertainty/calibrator.py

```diff
@@ -68,13 +70,59 @@ def apply_calibration(self, preds, uncs) -> Tensor:
 
 @UncertaintyCalibratorRegistry.register("platt")
 class PlattCalibrator(UncertaintyCalibrator):
-    def calibrate(self, preds, uncs, targets, mask) -> Tensor:
-        ...
-        return
-
-    def apply_calibration(self, preds, uncs) -> Tensor:
-        ...
-        return
+    """
+    A calibration method for classification datasets based on the Platt scaling algorithm.
+    As discussed in https://arxiv.org/abs/1706.04599 and Platt, J. (2000). Probabilistic outputs for
+    support vector machines and comparison to regularized likelihood methods. In A. Smola, P. 
+    Bartlett, B. Schölkopf, & D. Schuurmans (Eds.), Advances in large margin classifiers. Cambridge:
+    MIT Press.
+    In Platt's paper, he suggests using the number of positive and negative examples in the dataset
+    used to train the model to adjust the value of target probabilities used to fit the parameters. 
+    """
+
+    def calibrate(self, preds, targets, mask, training_targets: None | Tensor = None) -> Tensor:
+        if not torch.logical_or((targets[mask] == 0), (targets[mask] == 1)).all():
+            raise ValueError("Platt scaling is only implemented for binary classification tasks.")
+
+        if training_targets is not None:
+            negative_targets = (1 / ((training_targets == 0).sum(dim=0) + 2)).expand_as(targets)
+            positive_targets = (
+                ((training_targets == 1).sum(dim=0) + 1) / ((training_targets == 1).sum(dim=0) + 2)
+            ).expand_as(targets)
+
+            targets = targets.float()
+            targets[targets == 0] = negative_targets[targets == 0]
+            targets[targets == 1] = positive_targets[targets == 1]
+
+        platt_a = []
+        platt_b = []
```

Comment:
can we just call these `a` and `b` in line with the terminology used in the original equations we cite?

Reply:
as far as name collisions go, you just can call these `self.a` and `self.b` to not overlap with the local function below